torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1341,6 +1343,274 @@ def __init__(self, var_to_sources):
         super().__init__(var_to_sources, lambda n: n.name(), var_to_sources)
 
 
+class DynamicDimConstraintPrinter(StrPrinter):
+    """
+    Printer for dynamic dim constraints.
+    - Instead of t.size()[d] it prints dynamic_dim(t, d)
+    - Instead of Eq(_, _), Mod(_, _), etc. it prints _ == _, _ % _, etc.
+
+    We use this to suggest code for specifying dynamic dim constraints.
+    """
+    def __init__(self, symbol_to_source):
+        super().__init__()
+        self.symbol_to_source = symbol_to_source
+
+    def print_source(self, source) -> str:
+        return f"dynamic_dim({source.base.name()}, {source.idx})"
```

Comment:
This is going to print L['x'] instead of x, I think

Reply:
With export, you kind of want to do a rewrite, but not in the way you expect. Suppose you have

```
def f(x, y):
  ...

inp = [torch.randn(3), torch.randn(3)]
torch._dynamo.export(f, inp)
```

for the result to be directly actionable, you have to directly say `dynamic_dim(inp[0], 0)` e.g.

Your current message falls in an uncanny valley, where it sort of looks like it is copy pastable, but it's not. It's better to go all the way, or not at all.